### ISSUE TITLE
b24fa13 broke nzbget post processor script (I think)

### ISSUE THREAD

**Issue by bleomycin** [nosmokingbandit/watcher#160]
Started by applying fix from https://github.com/nosmokingbandit/watcher/pull/145 which worked great before b24fa13:

The script is now failing with:

```
Sat Jan 14 16:38:51 2017        INFO    watcher: Traceback (most recent call last):
Sat Jan 14 16:38:51 2017        INFO    watcher:   File "/home/nzbget/nzbget/scripts/watcher.py", line 61, in <module>
Sat Jan 14 16:38:51 2017        INFO    watcher:     response = json.loads(urllib2.urlopen(request, timeout=600).read())
Sat Jan 14 16:38:51 2017        INFO    watcher:   File "/usr/lib/python2.7/urllib2.py", line 154, in urlopen
Sat Jan 14 16:38:51 2017        INFO    watcher:     return opener.open(url, data, timeout)
Sat Jan 14 16:38:51 2017        INFO    watcher:   File "/usr/lib/python2.7/urllib2.py", line 437, in open
Sat Jan 14 16:38:51 2017        INFO    watcher:     response =   ##meth(req, response)
Sat Jan 14 16:38:51 2017        INFO    watcher:   File "/usr/lib/python2.7/urllib2.py", line 550, in http_response
Sat Jan 14 16:38:51 2017        INFO    watcher:     'http', request, response, code, msg, hdrs)
Sat Jan 14 16:38:51 2017        INFO    watcher:   File "/usr/lib/python2.7/urllib2.py", line 475, in error
Sat Jan 14 16:38:51 2017        INFO    watcher:     return self._call_chain(*args)
Sat Jan 14 16:38:51 2017        INFO    watcher:   File "/usr/lib/python2.7/urllib2.py", line 409, in _call_chain
Sat Jan 14 16:38:51 2017        INFO    watcher:     result = func(*args)
Sat Jan 14 16:38:51 2017        INFO    watcher:   File "/usr/lib/python2.7/urllib2.py", line 558, in http_error_default
Sat Jan 14 16:38:51 2017        INFO    watcher:     raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
Sat Jan 14 16:38:51 2017        INFO    watcher: urllib2.HTTPError: HTTP Error 500: Internal Server Error
```

**Comment by vertigo235** [comment]
Duplicate of issue https://github.com/nosmokingbandit/watcher/issues/142

**Comment by bleomycin** [comment]
Sorry this is still failing for me, updated to d6ce824, grabbed latest post processing script. I'm using the [proxy] feature if it makes any difference, but i have the appropriate settings in nzbget according to the wiki to compensate. I also tested with proxy set to false and settings changed in nzbget to reflect, no change. Debian Jessie, Mono version 4.6.2

```
Sun Jan 15 12:31:26 2017        INFO    watcher: Traceback (most recent call last):
Sun Jan 15 12:31:26 2017        INFO    watcher:   File "/home/nzbget/nzbget/scripts/watcher.py", line 61, in <module>
Sun Jan 15 12:31:26 2017        INFO    watcher:     response = json.loads(urllib2.urlopen(request, timeout=600).read())
Sun Jan 15 12:31:26 2017        INFO    watcher:   File "/usr/lib/python2.7/urllib2.py", line 154, in urlopen
Sun Jan 15 12:31:26 2017        INFO    watcher:     return opener.open(url, data, timeout)
Sun Jan 15 12:31:26 2017        INFO    watcher:   File "/usr/lib/python2.7/urllib2.py", line 437, in open
Sun Jan 15 12:31:26 2017        INFO    watcher:     response = meth(req, response)
Sun Jan 15 12:31:26 2017        INFO    watcher:   File "/usr/lib/python2.7/urllib2.py", line 550, in http_response
Sun Jan 15 12:31:26 2017        INFO    watcher:     'http', request, response, code, msg, hdrs)
Sun Jan 15 12:31:26 2017        INFO    watcher:   File "/usr/lib/python2.7/urllib2.py", line 475, in error
Sun Jan 15 12:31:26 2017        INFO    watcher:     return self._call_chain(*args)
Sun Jan 15 12:31:26 2017        INFO    watcher:   File "/usr/lib/python2.7/urllib2.py", line 409, in _call_chain
Sun Jan 15 12:31:26 2017        INFO    watcher:     result = func(*args)
Sun Jan 15 12:31:26 2017        INFO    watcher:   File "/usr/lib/python2.7/urllib2.py", line 558, in http_error_default
Sun Jan 15 12:31:26 2017        INFO    watcher:     raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
Sun Jan 15 12:31:26 2017        INFO    watcher: urllib2.HTTPError: HTTP Error 500: Internal Server Error
Sun Jan 15 12:31:26 2017        ERROR   Post-process-script watcher.py for something.Watcher failed (terminated with unknown status)
```

**Comment by bleomycin** [comment]
Here are server-side logs if they are helpful at all as well:

```
INFO 2017-01-15 15:55:47,408 core.postprocessing.POST: #################################
INFO 2017-01-15 15:55:47,408 core.postprocessing.POST: Post-processing request received.
INFO 2017-01-15 15:55:47,408 core.postprocessing.POST: #################################
INFO 2017-01-15 15:55:47,408 core.postprocessing.get_filename: Finding movie file name.
INFO 2017-01-15 15:55:47,410 core.postprocessing.get_filename: Post-processing file /media/raid/incoming/watcher/something.2013.1080p.BluRay.x264.YIFY.Watcher/something.2013.1080p.BluRay.x264.YIFY.mp4.
INFO 2017-01-15 15:55:47,411 core.postprocessing.POST: Parsing release name for information.
INFO 2017-01-15 15:55:47,411 core.postprocessing.POST: Gathering release information.
INFO 2017-01-15 15:55:47,411 core.postprocessing.get_movie_info: Searching local database for guid.
INFO 2017-01-15 15:55:47,411 core.sqldb.get_single_search_result: Retreving search result details for https://api.nzb.su/getnzb/removed.
INFO 2017-01-15 15:55:47,412 core.postprocessing.get_movie_info: Searchresult found by guid.
INFO 2017-01-15 15:55:47,412 core.postprocessing.get_movie_info: Searching local database by imdbid.
INFO 2017-01-15 15:55:47,412 core.sqldb.get_movie_details: Retreving details for tt1411250.
INFO 2017-01-15 15:55:47,413 core.postprocessing.get_movie_info: Movie data found locally by imdbid.
INFO 2017-01-15 15:55:47,413 core.postprocessing.POST: Post-processing as Complete.
INFO 2017-01-15 15:55:47,413 core.postprocessing.complete: Marking guid as Finished.
INFO 2017-01-15 15:55:47,414 core.updatestatus.searchresults: Marking guid https://api.nzb.su/getnzb/removed as Finished.
INFO 2017-01-15 15:55:47,414 core.updatestatus.searchresults: Marking https://api.nzb.su/getnzb/removed as Finished in SEARCHRESULTS.
INFO 2017-01-15 15:55:47,414 core.sqldb.update: Updating https://api.nzb.su/getnzb/removed to Finished in SEARCHRESULTS.
INFO 2017-01-15 15:55:47,416 core.updatestatus.searchresults: Successfully marked https://api.nzb.su/getnzb/removed as Finished in SEARCHRESULTS.
INFO 2017-01-15 15:55:47,417 core.updatestatus.markedresults: Marking https://api.nzb.su/getnzb/removed as Finished in MARKEDRESULTS.
INFO 2017-01-15 15:55:47,417 core.sqldb.update: Updating https://api.nzb.su/getnzb/removed to Finished in MARKEDRESULTS.
INFO 2017-01-15 15:55:47,418 core.updatestatus.markedresults: Successfully marked https://api.nzb.su/getnzb/removed as Finished in MARKEDRESULTS.
INFO 2017-01-15 15:55:47,419 core.postprocessing.complete: Setting MOVIE status.
INFO 2017-01-15 15:55:47,419 core.sqldb.get_distinct: Getting distinct values for tt1411250 in SEARCHRESULTS
INFO 2017-01-15 15:55:47,419 core.updatestatus.movie_status: Setting MOVIES tt1411250 status to Finished.
INFO 2017-01-15 15:55:47,419 core.sqldb.update: Updating tt1411250 to Finished in MOVIES.
ERROR 2017-01-15 15:55:47,421 cherrypy.error.140196028673936.error: [15/Jan/2017:15:55:47] HTTP 
Traceback (most recent call last):
  File "/opt/watcher/lib/cherrypy/_cprequest.py", line 670, in respond
    response.body = self.handler()
  File "/opt/watcher/lib/cherrypy/lib/encoding.py", line 220, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/opt/watcher/lib/cherrypy/_cpdispatch.py", line 60, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/opt/watcher/core/postprocessing.py", line 93, in POST
    response = json.dumps(self.complete(data), indent=2, sort_keys=True)
  File "/opt/watcher/core/postprocessing.py", line 534, in complete
    self.sql.update('MOVIES', 'finished_date', result['data']['finisheddate'],
KeyError: 'finisheddate'
INFO 2017-01-15 15:55:47,421 cherrypy.error.140196028673936.error: [15/Jan/2017:15:55:47] HTTP 
Request Headers:
  HOST: 192.168.2.9:9090
  CONNECTION: close
  Remote-Addr: 192.168.2.7
  Content-Length: 287
  USER-AGENT: Mozilla/5.0
  Content-Type: application/x-www-form-urlencoded
  ACCEPT-ENCODING: identity
```

**Comment by WilliamJBrown** [comment]
Also having this problem myself too. 

`INFO 2017-01-16 02:43:08,858 core.sqldb.update: Updating tt0144084 to Finished in MOVIES.
ERROR 2017-01-16 02:43:08,859 cherrypy.error.140270509412816.error: [16/Jan/2017:02:43:08] HTTP 
Traceback (most recent call last):
  File "/opt/watcher/app/lib/cherrypy/_cprequest.py", line 670, in respond
    response.body = self.handler()
  File "/opt/watcher/app/lib/cherrypy/lib/encoding.py", line 220, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/opt/watcher/app/lib/cherrypy/_cpdispatch.py", line 60, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/opt/watcher/app/core/postprocessing.py", line 93, in POST
    response = json.dumps(self.complete(data), indent=2, sort_keys=True)
  File "/opt/watcher/app/core/postprocessing.py", line 534, in complete
    self.sql.update('MOVIES', 'finished_date', result['data']['finisheddate'],
KeyError: 'finisheddate'
INFO 2017-01-16 02:43:08,859 cherrypy.error.140270509412816.error: [16/Jan/2017:02:43:08] HTTP 
Request Headers:
  HOST: 192.168.1.190:9090
  CONNECTION: close
  Remote-Addr: 172.17.42.1
  Content-Length: 318
  USER-AGENT: Mozilla/5.0
  Content-Type: application/x-www-form-urlencoded
  ACCEPT-ENCODING: identity`

**Comment by nosmokingbandit** [comment]
This was fixed in e042f1d7b210da9f26d6f95e1e3ca52733938dc3.
